t/process-type-privileged-agent-connections.t

```diff
@@ -0,0 +1,292 @@
+# vim:set ft= ts=4 sw=4 et fdm=marker:
+
+use lib '.';
+use t::TestCore;
+
+master_process_enabled(1);
+
+repeat_each(1);
+
+plan tests => repeat_each() * (blocks() * 5);
+
+
+$ENV{TEST_NGINX_LUA_PACKAGE_PATH} = $t::TestCore::lua_package_path;
+
+run_tests();
+
+__DATA__
+
+=== TEST 1: specify connections to enable_privileged_agent
+--- http_config
+    lua_package_path "$TEST_NGINX_LUA_PACKAGE_PATH";
+
+    init_by_lua_block {
+        local process = require "ngx.process"
+        local ok, err = process.enable_privileged_agent(10)
+        if not ok then
+            ngx.log(ngx.ERR, "enable_privileged_agent failed: ", err)
+        end
+    }
+
+    init_worker_by_lua_block {
+        local base = require "resty.core.base"
+        local typ = (require "ngx.process").type()
+
+        if typ == "privileged agent" then
+            ngx.log(ngx.WARN, "process type: ", typ)
+            ngx.timer.at(0, function()
+                local tcpsock = ngx.socket.tcp()
+                local ok, err = tcpsock:connect("127.0.0.1", 1984)
+
+                if ok then
+                    ngx.log(ngx.INFO, "connect ok ")
+                else
+                    ngx.log(ngx.INFO, "connect not ok " .. tostring(err))
+                end
+            end)
+        end
+    }
+--- config
+    location = /t {
+        content_by_lua_block {
+            ngx.sleep(0.1)
+            local typ = require "ngx.process".type()
+            ngx.say("type: ", typ)
+        }
+    }
+--- request
+GET /t
+--- response_body
+type: worker
+--- error_log
+connect ok
+--- no_error_log
+connect not ok
+enable_privileged_agent failed
+--- skip_nginx: 5: < 1.11.2
+--- wait: 0.2
+
+
+
+=== TEST 2: connections exceed limit
+--- http_config
+    lua_package_path "$TEST_NGINX_LUA_PACKAGE_PATH";
+
+    init_by_lua_block {
+        local process = require "ngx.process"
+        local ok, err = process.enable_privileged_agent(10)
+        if not ok then
+            ngx.log(ngx.ERR, "enable_privileged_agent failed: ", err)
+        end
+    }
+
+    init_worker_by_lua_block {
+        local base = require "resty.core.base"
+        local typ = (require "ngx.process").type()
+
+        if typ == "privileged agent" then
+            ngx.log(ngx.WARN, "process type: ", typ)
+            ngx.timer.at(0, function()
+
+                for i = 1, 10 do
+                    local tcpsock = ngx.socket.tcp()
```

Comment:
reach limit, but not exceed. why failed?

Reply:
i found openresty will create a little socket when you start. but i did'nt deep research.